### ISSUE TITLE
Memory Corruption Bug or My Mistake?

### ISSUE THREAD

**Issue by Soulsbane** [vlang/v#5595]
**V version:** 1.28 & custom compile
**OS:** Manjaro Linux 5.4

**What did you do?**
I have a function that creates a CLI choice menu out of a passed array. It returns the array value based on what choice number the that the user choose. I've attached the code that works in 1.27 but not in 1.28 release or a version compiled myself.

The test code and result  look like this:
`
Matches len = 2
1) 
2) �W��1V
What word? 2
Choice: 2

`
The test code and result should look like this:

`
Matches len = 2
1) Matrix
2) Matrix Revolutions
What word? 2
Choice: Matrix Revolutions
`
I hope I've adequately explained that. 
[app.zip](https://github.com/vlang/v/files/4856559/app.zip)

**Comment by SleepyRoy** [comment]
I tried mingw64 and tcc as backend on windows. For both, '2) Matrix Revolutions' is nicely printed but '1) Matrix' is not.  
And let me paste your code here so more people could have a look.
 
```V
import os
import strings

fn prompt(message string) string {
	print(message)
	answer := os.get_line().trim_space()

	return answer.trim('\n')
}

fn choice_menu(message string, choices []string) string {
	for i, choice in choices {
		option_num := i + 1
		println("$option_num) $choice")
	}

	choice_num := prompt(message).int() - 1

	if choice_num >= choices.len || choice_num < 0 {
		return ""
	}

	return choices[choice_num]
}

struct JotzCollection {
	mut:
		entries map[string]string
		title_size int = 1
}

fn (mut collection JotzCollection) add(name string, note string) {
	if name !in collection.entries {
		collection.entries[name] = note
	}
}

fn (collection JotzCollection) find(noteName string, caseSensitive bool) []string {
	mut matches := []string{}

	for name, _ in collection.entries {
		if caseSensitive {
			if name.contains(noteName) {
				matches << name
			}
		}
		else {
			if name.to_lower().contains(noteName.to_lower()) {
				matches << name
			}
		}
	}

	return matches
}

fn main() {
	mut collection := JotzCollection{}
	collection.add("Matrix", "Was a great move")
	collection.add("Matrix Revolutions", "Was Meh")
	collection.add("Parasite", "Wow what a great movie!")

	matches := collection.find("Matrix", false)
	println("Matches len = $matches.len")
	choice := choice_menu("Which Match? ", matches)
	println("Choice: $choice")
}
```

**Comment by medvednikov** [comment]
Interesting, can confirm something is wrong.

**Comment by medvednikov** [comment]
Ok I know what this is, a bug with map keys.

This fixes it for now:

```
matches << name.clone()
```

**Comment by spytheman** [comment]
This is fixed (or masked) by ca1f47a .

**Comment by medvednikov** [comment]
Indeed, this was just fixed by @spytheman
